engine/particle/src/particle.cpp

```diff
@@ -1314,12 +1316,9 @@ namespace dmParticle
 
         if (j < particle_count)
         {
-            if (emitter->m_RenderWarning == 0)
-            {
-                res = GENERATE_VERTEX_DATA_MAX_PARTICLES_EXCEEDED;
-                emitter->m_RenderWarning = 1;
-            }
+            res = GENERATE_VERTEX_DATA_MAX_PARTICLES_EXCEEDED;
```

Comment:
I'm unsure this is useful since the engine can handle the case and create small cpu size and we can handle any number of particules gpu side. The solve to that is to produce fewer particles.

Reply:
> and we can handle any number of particules gpu side.

The point with our numbers is that the developer should have control over their budgets.

True, you've minimized CPU budget (and that should probably be a new setting in the game.project),
but, the GPU budget still stands.

With this change, you change how we work in general with memory.

> The solve to that is to produce fewer particles.

The user will not realize that they're not seeing all their particles anymore if they exceed the limit.
The particles will just silently be dropped, and that's not ok.

Reply:
Just to be clear, with this patch they are not silently dropped. However we discussed offline and agreed to add a cap on the GPU side buffer and then return to having them silently dropped (after a single warning).